azurerm/internal/services/loganalytics/log_analytics_clusters_resource.go

```diff
@@ -0,0 +1,397 @@
+	}
+	d.Set("name", id.Name)
+	d.Set("resource_group_name", id.ResourceGroup)
+	d.Set("location", location.NormalizeNilable(resp.Location))
+	if err := d.Set("identity", flattenArmLogAnalyticsIdentity(resp.Identity)); err != nil {
+		return fmt.Errorf("setting `identity`: %+v", err)
+	}
+	if props := resp.ClusterProperties; props != nil {
+		if err := d.Set("key_vault_property", flattenArmLogAnalyticsKeyVaultProperties(props.KeyVaultProperties)); err != nil {
+			return fmt.Errorf("setting `key_vault_property`: %+v", err)
+		}
+		d.Set("cluster_id", props.ClusterID)
+	}
+	d.Set("size_gb", flattenArmLogAnalyticsClusterSku(resp.Sku))
+
+	return tags.FlattenAndSet(d, resp.Tags)
+}
+
+func resourceArmLogAnalyticsClusterUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).LogAnalytics.ClusterClient
+	ctx, cancel := timeouts.ForUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.LogAnalyticsClusterID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	parameters := operationalinsights.ClusterPatch{
+		ClusterPatchProperties: &operationalinsights.ClusterPatchProperties{},
+	}
+
+	if d.HasChange("key_vault_property") {
+		parameters.ClusterPatchProperties.KeyVaultProperties = expandArmLogAnalyticsClusterKeyVaultProperties(d.Get("key_vault_property").([]interface{}))
+	}
+
+	if d.HasChange("size_gb") {
+		parameters.Sku = &operationalinsights.ClusterSku{
+			Capacity: utils.Int64(int64(d.Get("size_gb").(int))),
+			Name:     operationalinsights.CapacityReservation,
+		}
+	}
+
+	if d.HasChange("tags") {
+		parameters.Tags = tags.Expand(d.Get("tags").(map[string]interface{}))
+	}
+
+	if _, err := client.Update(ctx, id.ResourceGroup, id.Name, parameters); err != nil {
+		return fmt.Errorf("updating Log Analytics Cluster %q (Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
+	}
+
+	// Need to wait for the cluster to actually finish updating the resource before continuing
+	// since the service returns a 200 instantly while it's still updating in the background
+	log.Printf("[INFO] Checking for Log Analytics Cluster provisioning state")
+
+	updateWait := &resource.StateChangeConf{
+		Pending:    []string{string(operationalinsights.Updating)},
+		Target:     []string{string(operationalinsights.Succeeded)},
+		MinTimeout: 1 * time.Minute,
+		Timeout:    d.Timeout(schema.TimeoutUpdate),
+		Refresh: func() (interface{}, string, error) {
+			log.Printf("[INFO] checking on state of Log Analytics Cluster %q", id.Name)
+
+			resp, err := client.Get(ctx, id.ResourceGroup, id.Name)
+			if err != nil {
+				return nil, "nil", fmt.Errorf("polling for the status of Log Analytics Cluster %q (Resource Group %q): %v", id.Name, id.ResourceGroup, err)
+			}
+
+			if resp.ClusterProperties.ProvisioningState != operationalinsights.Updating && resp.ClusterProperties.ProvisioningState != operationalinsights.Succeeded {
```

Comment:
This will need nil checking as `ClusterProperties` is a pointer.

Reply:
Fixed.